### PR TITLE
Let backends decide root change id and change id length

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: tests/test_log_command.rs [hunk 1]
```diff
@@ -295,7 +295,7 @@ fn test_log_prefix_highlight_brackets() {
       "Change " change_id.shortest_prefix_and_brackets({0}) " " description.first_line()
       " " commit_id.shortest_prefix_and_brackets({0}) " " branches
     "#,
-            len.map(|l| l.to_string()).unwrap_or(String::default())
+            len.map(|l| l.to_string()).unwrap_or_default()
```
Comment (yuja) [inline tests/test_log_command.rs:298]: nit: unrelated change

**Review by yuja — APPROVED** [review]

File: tests/test_log_command.rs (hunk 1 above)
Comment (martinvonz) [reply tests/test_log_command.rs]: Will revert
